.ci/jenkins/Jenkinsfile.prod.nightly

```
@@ -34,34 +36,43 @@ pipeline {
+
+                    sh "npm list -g | grep build-chain"
+                }
+            }
+        }
         stage('Clone build configuration repo') {
             steps {
                 script {
-                    def currentBranch = env.CONFIG_BRANCH ?: env.DEFAULT_CONFIG_BRANCH ?: env.BRANCH_NAME ?: env.GIT_BRANCH
+                    def currentBranch = getBuildConfigBranch()
                     println "Checking out ${env.BUILD_CONFIGURATION_REPO_URL}:${currentBranch} into build_config folder"
                     sh "git clone -b ${currentBranch} --single-branch ${env.BUILD_CONFIGURATION_REPO_URL} build_config"
+
+                    // export PME parameters and build scripts from nightly build config
+                    def buildConfigAdditionalVariables = [
+                        'productVersion': env.PRODUCT_VERSION,
+                        'droolsProductVersion': env.DROOLS_PRODUCT_VERSION
+                    ]
+                    pmebuild.parseBuildConfig("$WORKSPACE/build_config/rhbop/nightly", buildConfigAdditionalVariables)
                 }
             }
         }
         stage('Build Nightly projects') {
             steps {
-                catchError(buildResult: 'UNSTABLE', stageResult: 'FAILURE')
-                {
-                    script {
-                        def pipelineHelper = new PipelineHelper(this)
-                        pipelineHelper.retry(
-                        {
-                            def SETTINGS_XML_ID = '5d9884a1-178a-4d67-a3ac-9735d2df2cef'
-                            def projectCollection = ['drools', 'optaplanner', 'optaplanner-quickstarts', 'jboss-integration/rhbop-optaplanner']
-                            def projectVariableMap = ['kiegroup_optaplanner': 'optaplannerVersion', 'kiegroup_drools': 'droolsProductVersion']
-                            def additionalVariables = [
-                                'productVersion': env.PRODUCT_VERSION,
-                                'droolsProductVersion': env.DROOLS_PRODUCT_VERSION,
-                                'drools-scmRevision': getCurrentBranch(),
-                                'optaplanner-quickstarts-scmRevision': getOptaPlannerQuickstartsBranch()
-                            ]
-                            pmebuild.buildProjects(projectCollection, "${SETTINGS_XML_ID}", "$WORKSPACE/build_config/rhbop/nightly", "${env.PME_CLI_PATH}", projectVariableMap, additionalVariables, [:])
-                        }, 2, 180*60)
+                script {
+                    withCredentials([string(credentialsId: "kie-ci1-token", variable: 'GITHUB_TOKEN')]) {
+                        def SETTINGS_XML_ID = '5d9884a1-178a-4d67-a3ac-9735d2df2cef'
+                        def buildBranch = getBuildBranch()
+                        def definitionFile = "https://raw.githubusercontent.com/${env.DEFINITION_FILE_OWNER}/optaplanner/${env.DEFINITION_FILE_BRANCH}/.ci/nightly-build-config.yaml"
```

what if we use `${GROUP:kiegroup}` and `${BRANCH:main}` place holders see https://github.com/kiegroup/build-chain-configuration-reader#url-format

```suggestion
                        def definitionFile = "https://raw.githubusercontent.com/${GROUP:kiegroup}/optaplanner/${BRANCH:main}/.ci/nightly-build-config.yaml"
```

---

this way we don't need `prod_cron_meta_nightly_pipeline.groovy rhbopNightlyStage`  refactoring

---

I agree

---

changes applied